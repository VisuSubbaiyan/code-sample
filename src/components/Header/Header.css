.header {
  background: #05164d;
  padding: 10px;
  color: #cccccc;
  display: flex;
  align-items: center;
}
.header .fa {
  font-size: 36px;
}

.header h1.heading {
  margin: 0;
  padding: 10px 16px;
}