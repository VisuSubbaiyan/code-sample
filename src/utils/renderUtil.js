import React from 'react';

import {Airport, BoardingPass, Ground} from '../components';

export const renderItems = items => items.map(item => {
  const { type, segment_id } = item;

  switch (type) {
    case 'GROUND': {
      return <Ground {...item} key={segment_id} className="segment-active no-top-padding" />;
    }
    
    case 'AIRPORT': {
      return <Airport {...item} key={segment_id} />;
    }

    default:
      return <BoardingPass {...item} key={segment_id} />
  }
});
