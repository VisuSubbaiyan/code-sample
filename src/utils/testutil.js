import { mount, shallow } from 'enzyme';
import { prettyPrint } from 'html';

export const mockFunction = jest.fn(value => value);

export const sampleFriend = {
  name: 'Sample friend',
  gender: 'male'
};

export const sampleId = 1;

export const createMountedComponent = ({ children, isShallow }) => {
  const mountedComponent = isShallow ? shallow(children) : mount(children);
  mountedComponent.htmlRender = () => prettyPrint(mountedComponent.html(), { indent_size: 2 });

  return mountedComponent;
};

