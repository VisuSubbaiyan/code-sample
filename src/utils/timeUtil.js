import moment from 'moment';

export const formatTime = time => {
  const tempTime = moment.duration(time);
  
  return `${tempTime.hours()}: ${tempTime.minutes()}`;
}