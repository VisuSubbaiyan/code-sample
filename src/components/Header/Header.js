import React from 'react';
import PropTypes from 'prop-types';

import IconButton from '../IconButton';
import './Header.css';

const Header = props => (
  <header className="header">
    <IconButton icon="angle-left" onClick={() => null} />
    <h1 className="heading">{props.heading}</h1>
  </header>
);

Header.propTypes = {
  heading: PropTypes.string.isRequired
}


export default Header;
