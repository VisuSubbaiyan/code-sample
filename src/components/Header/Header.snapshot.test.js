import React from 'react';

import Header from './Header';
import { createMountedComponent } from '../../utils/testutil';

describe('<Header /> should', () => {
  const mountedComponent = createMountedComponent({
    children: <Header heading="sample Text" />
  });

  const icon = mountedComponent.find('i');
  const heading = mountedComponent.find('h1');

  it('render correctly according to props', () => {
    expect(mountedComponent.htmlRender()).toMatchSnapshot();
    expect(icon).toHaveLength(1);
    expect(heading).toHaveLength(1);
  })
});