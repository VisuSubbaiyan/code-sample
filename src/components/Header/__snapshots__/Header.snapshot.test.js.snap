// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly according to props 1`] = `
"<header class=\\"header\\">
  <div><i class=\\"fa fa-angle-left\\"></i>
  </div>
  <h1 class=\\"heading\\">sample Text</h1>
</header>"
`;
